### DISCUSSION TITLE
Control Light by Surface Type?

### DISCUSSION THREAD

**Discussion by cmeid** [maca88/SmartBikeLights#37] [Q&A] [answered]
First off - what killer software @maca88! I really love how you implemented this.

I do have a question though: I ride a lot of mixed surfaces and gravel. Most of my rides are like 60% offroad and 40% on roads. I would love to be able to have the Varia automatically turn off two minutes after leaving a paved surface and turn back on when I go back onto a paved surface (in addition to all of the control you already give me)

Have you ever considered something like that? I have no idea whether it would be possible given the data available to ConnectIQ apps, but it would be absolutely amazing.

Best regards from Berlin!

**Comment by maca88** [comment, marked as answer]
Unfortunately, there is currently no way to detect whether you are riding on a paved or gravel surface, as no API is available for retrieving that. So for now, the only thing you can do is to use the "Position" filter, mark areas with different surfaces, and set a light mode for each.
